### PR TITLE
Add reduce option to gaussian_kl_divergence

### PR DESCRIPTION
**Pull request by delta2323**
This PR add `reduce` option to `F.gaussian_kl_divergence` following the design discussed in #2325.

### CODE REVIEW

File: tests/chainer_tests/functions_tests/loss_tests/test_vae.py [hunk 1]
```diff
@@ -18,15 +22,19 @@ def setUp(self):
...
+        if self.reduce == 'sum':
+            self.expect = numpy.sum(loss)
+        elif self.reduce == 'no':
+            self.expect = loss
 
     def check_gaussian_kl_divergence(self, mean, ln_var):
         m = chainer.Variable(mean)
         v = chainer.Variable(ln_var)
-        actual = cuda.to_cpu(vae.gaussian_kl_divergence(m, v).data)
+        actual = vae.gaussian_kl_divergence(m, v, self.reduce)
```
Comment (unnonouno) [inline tests/chainer_tests/functions_tests/loss_tests/test_vae.py]: `functions.gaussian_kl_divergence`?

File: chainer/functions/loss/vae.py [hunk 2]
```diff
@@ -21,23 +21,41 @@ def gaussian_kl_divergence(mean, ln_var):
...
+    the option `reduce`. If it is `'no'`, it holds the elementwise
+    loss values. If it is `'sum'`, loss values are summed up.
+
     Args:
         mean (~chainer.Variable): A variable representing mean of given
             gaussian distribution, :math:`\\mu`.
         ln_var (~chainer.Variable): A variable representing logarithm of
             variance of given gaussian distribution, :math:`\\log(\\sigma^2)`.
+        recude (str): Reduction option. Its value must be either
+            `'sum'` or `'no'`. Otherwise, `ValueError` is raised.
```
Comment (unnonouno) [inline chainer/functions/loss/vae.py]: <pre>:class:`ValueError`</pre> ?

File: chainer/functions/loss/vae.py [hunk 3]
```diff
@@ -21,23 +21,41 @@ def gaussian_kl_divergence(mean, ln_var):
     where :math:`S` is a diagonal matrix such that :math:`S_{ii} = \\sigma_i^2`
     and :math:`I` is an identity matrix.
 
+    The output is a varialbe whose value depends on the value of
+    the option `reduce`. If it is `'no'`, it holds the elementwise
```
Comment (unnonouno) [inline chainer/functions/loss/vae.py]: Use two backquotes

File: tests/chainer_tests/functions_tests/loss_tests/test_vae.py (hunk 1 above)
Comment (delta2323) [inline tests/chainer_tests/functions_tests/loss_tests/test_vae.py]: As other functions in this file also have this problem, I fixed and sent it as another PR (#2541)

File: chainer/functions/loss/vae.py (hunk 3 above)
Comment (delta2323) [inline chainer/functions/loss/vae.py]: Fixed

File: chainer/functions/loss/vae.py (hunk 2 above)
Comment (delta2323) [inline chainer/functions/loss/vae.py]: Fixed

**Comment by delta2323** [comment]
I also fixed #2540 in this PR.

**Comment by unnonouno** [comment]
LGTM

**Comment by unnonouno** [comment]
Rebase the master

**Comment by unnonouno** [comment]
Check test result

**Comment by unnonouno** [comment]
LGTM
